tracing/tests/support/span.rs

```diff
@@ -34,7 +34,6 @@ impl MockSpan {
                 name: Some(name.into()),
                 ..self.metadata
             },
-            ..self
```

Comment:
I'm surprised this compiles — is there an explanation of why it can be removed (perhaps in the clippy docs)? I'm just curious.

Reply:
I'll go back and check and provide the exact output when I make the other suggested changes. IIRC it said something like the `self` expansion was completely redundant because all the fields had already been specified. There's only one field here (`metadata`) right?

Reply:
Here is the output from the lint:

```
error: struct update has no effect, all the fields in the struct have already been specified
  --> tracing-attributes/tests/../../tracing/tests/support/span.rs:37:15
   |
37 |             ..self
   |               ^^^^
   |
   = note: `-D clippy::needless-update` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_update
```